app/bundles/ProjectBundle/Controller/AjaxController.php

```diff
@@ -75,4 +109,17 @@ public function addProjectsAction(Request $request, ProjectModel $projectModel,
 
         return $this->sendJsonResponse(['projects' => $projectOptions]);
     }
+
+    private function createProjectIfNotExists(string $name): int
+    {
+        if ($project = $this->projectRepository->findOneBy(['name' => $name])) {
+            return $project->getId();
+        }
+
+        $project = new Project();
+        $project->setName($name);
+        $this->projectModel->saveEntity($project);
+
+        return $project->getId();
+    }
```

Comment:
```suggestion
    private function createProjectIfNotExists(string $name, ProjectModel $projectModel, ProjectRepository $projectRepository): int
    {
        if ($project = $projectRepository->findOneBy(['name' => $name])) {
            return $project->getId();
        }

        $project = new Project();
        $project->setName($name);
        $projectModel->saveEntity($project);

        return $project->getId();
    }
```
We can avoid overriding the constructor by passing `$projectModel` and `$projectRepository` in `getLookupChoiceAction` method